Movie-App-Group-7/Movie-App-Group-7/Core/Data/Remote Data Source/Network/MovieNetworkModel.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  MovieNetworkModel.swift
+//  Movie-App-Group-7
+//
+//  Created by Mohammad Azri on 20/10/22.
+//
+
+import Foundation
+
+enum MovieNetworkResult {
+    case success([Movie])
+    case failure(String)
+}
+
+enum DetailMovieNetworkResult {
+    case success(Movie?)
+    case failure(String)
+}
+
+protocol MovieNetworkModel {
+    func getMovies(completion: @escaping (MovieNetworkResult) -> ())
+    func getMovie(movieId: String, completion: @escaping (DetailMovieNetworkResult) -> ())
+}
+
+final class MovieDefaultNetworkModel : MovieNetworkModel {
+    func getMovies(completion: @escaping (MovieNetworkResult) -> ()) {
+        guard let url = URL(string: "https://ghibliapi.herokuapp.com/films") else {
+            DispatchQueue.main.async {
+                completion(.failure("Bad URL"))
+            }
+            return
+        }
```

Comment:
Jangan pake guard let buat unwrap URL disini, kenapa? karena klo salah disini itu developer mistake bukan codenya, jadi untuk mencegah hal tersebut lebih baik ini di force unwrap.
```suggestion
        let url = URL(string: "https://ghibliapi.herokuapp.com/films")!
```

Reply:
Mau nanya Bang. Semisalnya invalid URL, ini ga ngecrash Bang kalo di force unwrap?